### PR TITLE
Show referral fees history and total

### PR DESCRIPTION
**Pull request by l2succes**
- We now have the ability to see the referral fee history and total amount accrued for a given customer 

## Preview

Screen Shot 2022-03-15 at 6 44 06 PM

### CODE REVIEW

**Comment by tjshipe** [comment]
@l2succes I tested this on mainnet and it displayed the data similar to your screenshot. The only 2 issues there are the Referral ID column can be completely removed, and the Fee Earned column on the right is empty. 

I think it's empty because the wrong key is being used. It should display the `referral_fee_accrual` from the api response. Also the fee accrual amounts will be low so you can use `usdFormatter` and pass in 4 as the second argument to show more decimals.

**Comment by l2succes** [comment]
@tjshipe Got it, thanks for the review. I'll get that fixed and push an update

**Comment by l2succes** [comment]
@tjshipe I fixed the fields to pull in the correct data and removed that unused Referral ID column

File: pages/referral.tsx [hunk 1]
```diff
@@ -442,27 +434,33 @@ export default function Referral() {
...
+                                    shallow={true}
+                                  >
+                                    <a className="text-th-fgd-2 underline hover:text-th-fgd-3 hover:no-underline">
+                                      {acct}
+                                    </a>
+                                  </Link>
+                                </Td>
+                                <Td className="flex items-center justify-end">
+                                  {usdFormatter(ref.referral_fee_accrual)}
+                                </Td>
```
Comment (tjshipe) [inline pages/referral.tsx:460]: The second argument to `usdFormatter` should probably be 4 so we show more decimals in the table

**Review by tjshipe — CHANGES_REQUESTED** [review]
If you reduce the width of the browser window, you can see the mobile view. Looks like there is a typo from our original mockups on line 470, should be "fee-earned". Also, should call the same `usdFormatter` with 4 decimals on line 479 to display the fee on mobile widths.

File: pages/referral.tsx (hunk 1 above)
Comment (saml33) [reply pages/referral.tsx:460]: I think if we do this:

export const usdFormatter = (value, decimals = 2, currency = true) => {
  if (decimals === 0) {
    value = Math.abs(value)
  }
  const fractionDigits =
    value < 1 ? countLeadingZeros(value) + decimals : decimals
  const config = currency ? { style: 'currency', currency: 'USD' } : {}
  return new Intl.NumberFormat('en-US', {
    ...config,
    minimumFractionDigits: fractionDigits,
    maximumFractionDigits: fractionDigits,
  }).format(value)
}

We won't need to pass decimals to number < 1

**Comment by saml33** [comment]
Screen Shot 2022-03-18 at 1 11 57 pm

We should settle on a date/time format. I think we have at least 3 including what you have done. I think your date format is the best as dd/mm/yyyy is prob confusing for USA. I would have your date format with the time on a new line as in the screenshot. What do you guys think?

**Comment by saml33** [comment]
Screen Shot 2022-03-18 at 1 09 14 pm

As Tyler mentioned, a few issues on mobile. You can remove pb-4 from this set of classes 'grid grid-flow-row grid-cols-2 gap-4 pb-4'

There is a key error in the console

**Comment by l2succes** [comment]
@saml33 I agree, the date `18 Mar 2022` format with the time on a new line would look best. I'll make that change. Might make sense to extract this into a reusable component at some point

**Comment by l2succes** [comment]
Screen Shot 2022-03-18 at 2 42 08 PM
Screen Shot 2022-03-18 at 2 45 38 PM

**Pull request merged**